app/tools/skills/alloy_discovery.py

```
@@ -0,0 +1,330 @@
+
+    Caller passes e.g. base_elements=["Mo","Nb","Ta","W","Hf"] with
+    fractions={"Hf": 0.05, "Mo": 0.2375, "Nb": 0.2375, "Ta": 0.2375, "W": 0.2375}.
+    Returns {"Hf": 5, "Mo": 24, "Nb": 24, "Ta": 24, "W": 23} (deterministic
+    rounding-drift correction so the result always sums to n_atoms).
+    """
+    if abs(sum(fractions.values()) - 1.0) > 1e-3:
+        raise ValueError(
+            f"fractions must sum to 1.0 (got {sum(fractions.values()):.4f})"
+        )
+    scaled = {el: f * n_atoms for el, f in fractions.items()}
+    rounded = {el: int(round(v)) for el, v in scaled.items()}
```

Ruff (0.15.12)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)